commit 7726ee3130352283887dcf760df1a2b0dc2d91df
Author: Rui Loura <rui@joyent.com>
Date:   2019-07-23T16:44:27+00:00 (3 months ago)
    
    MANTA-4428 rust-sharkspotter should put each object on a newline in default output file

diff --git a/README.md b/README.md
index 7fd06bd..a03becb 100644
--- a/README.md
+++ b/README.md
@@ -28,11 +28,20 @@ OPTIONS:
     -s, --shark <STORAGE_ID>          Find objects that belong to this shark
 ```
 
+## Example
+```
+$ cargo run -- --domain east.joyent.us --shark 1.stor -M 1 -m 1
+$ json -f shard_1_1.stor.objs -g
+```
+
+__This can be a big file so [json](https://github.com/trentm/json) may struggle with it__
+
 ## Development
 
 Before integration run:
 ```
 cargo fmt
+cargo clippy
 ```
 
 and 
diff --git a/src/lib.rs b/src/lib.rs
index af09354..4771b21 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -99,7 +99,6 @@ where
     );
 
     match mclient.sql(query.as_str(), vec![], r#"{"timeout": 10000}"#, |a| {
-        let id: String = serde_json::from_value(a[0][id_name].clone()).unwrap();
         query_handler(a, shard_num, handler)
     }) {
         Ok(()) => Ok(()),
@@ -112,7 +111,7 @@ where
 
 fn iter_ids<F>(
     id_name: &str,
-    moray_socket: &String,
+    moray_socket: &str,
     conf: &config::Config,
     shard_num: u32,
     mut handler: F,
@@ -120,7 +119,7 @@ fn iter_ids<F>(
 where
     F: FnMut(MantaObject, u32) -> Result<(), Error>,
 {
-    let mut mclient = MorayClient::from_str(moray_socket.as_str()).unwrap();
+    let mut mclient = MorayClient::from_str(moray_socket).unwrap();
     let mut start_id = conf.begin;
     let mut end_id = conf.begin + conf.chunk_size - 1;
     let largest_id = match find_largest_id_value(&mut mclient, id_name) {
diff --git a/src/main.rs b/src/main.rs
index 0ba3649..ca12dad 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -40,6 +40,7 @@ fn main() -> Result<(), Error> {
         let file = file_map.get_mut(&shard).unwrap();
         let buf = serde_json::to_string(&mobj)?;
         file.write_all(buf.as_bytes())?; // TODO: match
+        file.write_all(b"\n")?;
 
         Ok(())
     })
